PKHeX.WinForms/MainWindow/Main.cs

```diff
@@ -3838,7 +3839,18 @@ private void getSlotFiller(int offset, PictureBox pb)
             pb.Image = sprite;
             pb.BackColor = Color.Transparent;
             pb.Visible = true;
-        }
+			checkIfLegalPKMSlot(p, pb);
+        }
+		private void checkIfLegalPKMSlot(PKM pk, PictureBox pb)
+		{
+			LegalityAnalysis la = new LegalityAnalysis(pk);
+			if (pk.GenNumber >= 6 && pk.Species != 0 && !la.Valid)
+			{
```

Comment:
Inconsistent use of curly braces.  The else section doesn't use them.  I personally would go with adding them to the else, but @kwsch (from what I've seen) would remove the curly braces here.